processor/src/main/java/com/linecorp/decaton/processor/runtime/internal/ProcessingContextImpl.java

```diff
@@ -98,16 +88,20 @@ public String subscriptionId() {
     }
 
     @Override
-    public DeferredCompletion deferCompletion() {
-        completionDeferred.set(true);
-        return completion;
+    public Completion deferCompletion(Supplier<Boolean> callback) {
+        if (deferredCompletion.get() == null) {
```

Comment:
By creating a `CompletionImpl`  instance every time? I don't think so.

Reply:
I wondered if we should check & set deferredCompletion atomically, since we could call `deferCompletion()` later from async-thread (i.e. ≠ processor thread), so it could cause a race.

But I realized that even in such case, deferCompletion should be called first in processor thread so current code will correctly do one-time initialization of the callback.

Okay, then fine